File: backend/src/main/java/com/rj/ecommerce_backend/product/service/CategoryService.java
package com.rj.ecommerce_backend.product.service;

import com.rj.ecommerce_backend.product.filters.CategoryCreateDTO;
import com.rj.ecommerce_backend.product.filters.CategoryResponseDTO;
import com.rj.ecommerce_backend.product.search.CategorySearchCriteria;
import com.rj.ecommerce_backend.product.filters.CategoryUpdateDTO;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.List;
import java.util.Optional;

public interface CategoryService {

    CategoryResponseDTO createCategory(CategoryCreateDTO categoryDTO);
    Optional<CategoryResponseDTO> getCategoryById(Long id);
    Page<CategoryResponseDTO> getAllCategories(Pageable pageable, CategorySearchCriteria criteria);
    List<String> getCategoryNames();
    CategoryResponseDTO updateCategory(Long id, CategoryUpdateDTO updatedCategoryDTO);
    void deleteCategory(Long id);

}

File: backend/src/main/kotlin/com/rj/ecommerce_backend/product/service/CategoryService.kt
package com.rj.ecommerce_backend.product.service

import com.rj.ecommerce.api.shared.dto.product.CategoryCreateRequestDTO
import com.rj.ecommerce.api.shared.dto.product.CategoryDTO
import com.rj.ecommerce.api.shared.dto.product.CategoryUpdateDTO
import com.rj.ecommerce_backend.product.search.CategorySearchCriteria
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable

interface CategoryService {
    fun createCategory(requestDTO: CategoryCreateRequestDTO): CategoryDTO
    fun getCategoryById(categoryId: Long): CategoryDTO
    fun getAllCategories(pageable: Pageable, criteria: CategorySearchCriteria): Page<CategoryDTO>
    fun getCategoryNames(): List<String>
    fun updateCategory(categoryId: Long, request: CategoryUpdateDTO): CategoryDTO
    fun deleteCategory(categoryId: Long)
}